### PR TITLE
precommit: add `pyproject`'s hooks

### PR DESCRIPTION
**Pull request by Borda**
Reaction to #1411. Since there are no `setup.cfg`, but there are `pyproject.toml` configs, it may be worse to validate them...

cc: @rbren, @li-boxuan

### CODE REVIEW

**Comment by li-boxuan** [comment]
Mind merging https://github.com/OpenDevin/OpenDevin/pull/1411 to this PR? There might be a conflict.

**Comment by Borda** [comment]
> Mind merging #1411 to this PR? There might be a conflict.

**Comment by li-boxuan** [comment]
Oh I meant by cherry-picking that commit to this branch. These changes are interconnected and might as well appear in the same PR. Just my personal opinion, though.

**Review by rbren — APPROVED** [review]

File: dev_config/python/.pre-commit-config.yaml [hunk 1]
```diff
@@ -7,6 +7,22 @@ repos:
...
+    rev: v0.16
+    hooks:
+      - id: validate-pyproject
+
+  - repo: https://github.com/asottile/setup-cfg-fmt
+    rev: v2.5.0
+    hooks:
+      - id: setup-cfg-fmt
+        always_run: true
+        pass_filenames: false
```
[the remark this answers is not in the document]
Comment (rbren) [reply dev_config/python/.pre-commit-config.yaml:24]: I think my merge added this back in

**Pull request merged**